homeassistant/components/zwave_js/services.py

```diff
@@ -48,6 +54,13 @@
 
 T = TypeVar("T", ZwaveNode, Endpoint)
 
+TARGET_VALIDATORS = {
+    vol.Optional(ATTR_AREA_ID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_LABEL_ID): vol.All(cv.ensure_list, [cv.string]),
```

Comment:
For the majority of these services, they are at the device level. But users want to be able to target an area or an entity for convenience. Perhaps the solution here is to create a selector like the target one but rolls up the ids to the level specified by the integration. e.g. If it's at the device level, provide all the different target modes, but just return a list of unique devices to the integration. That would make this simpler in the integration for sure while still helping users use those other target types

Reply:
Maybe. Targeting a area sounds reasonable, but using an entity when the appropriate id is a device sounds questionable.

Reply:
it looks like we've been using that pattern since the beginning (first service PR: https://github.com/home-assistant/core/pull/46673). While I agree from a UI perspective that device IDs are sufficient, for YAML automation people entity IDs are easier to write. I am pretty sure there was an extensive discussion about this in Discord. If we have changed our stance, that's fine, but we should be clear about that. We would also have to figure out how to handle it to prevent a breaking change